packages/manager/src/components/NotificationMenu/NotificationMenu.tsx

```diff
@@ -24,6 +26,35 @@ export interface Props {
   open: boolean;
 }
 
+const isNotificationEventObj = (
+  body: string | JSX.Element
+): body is JSX.Element => {
+  return typeof (body as JSX.Element) === typeof {};
+};
+
+/**
+ * Filter notifications based on a list of filtered events.
+ * NotificationItems may have a body containing a string or an event object.
+ */
+const filterNotifications = (eventNotifications: NotificationItem[]) => {
+  // Extract events from notification items.
+  const events = eventNotifications.map((notification: NotificationItem) => {
+    return isNotificationEventObj(notification.body)
+      ? notification.body?.props.event
+      : notification.body;
+  });
+  const filteredEvents = removeBlocklistedEvents(events);
+
+  // Include all notifications that are non-blocklisted events.
+  return eventNotifications.filter((notification) => {
+    return (
+      (isNotificationEventObj(notification.body) &&
+        filteredEvents.includes(notification.body.props.event)) ||
+      !isNotificationEventObj(notification.body)
+    );
+  });
```

Comment:
Nice to have:
We can remove explicit `return` from `filter`
```suggestion
  return eventNotifications.filter((notification) => 
      (isNotificationEventObj(notification.body) &&
        filteredEvents.includes(notification.body.props.event)) ||
      !isNotificationEventObj(notification.body)
    );
```